htdocs/core/modules/mailings/xinputfile.modules.php

```diff
@@ -179,7 +179,8 @@ function add_to_target($mailing_id,$filtersarray=array())
 					        {
 					        	$i++;
 					        	$langs->load("errors");
-					        	$this->error = $langs->trans("ErrorFoundBadEmailInFile",$i,$cpt,$email);
+							$msg = $langs->trans("ErrorFoundBadEmailInFile",$i,$cpt,$email);
+					        	if(!empty($msg)) $this->error = $msg;
```

Comment:
I can't image you get an empty error message here.
$msg is $langs->trans("..
So you should always have a string here (it may be wrong translated but you should get something).

So the problem is not that we show an error empty when we should not, but that the message is empty.
We should here show an error. At least we shold add a else here

if(!empty($msg)) $this->error = $msg;
else $this->error = 'ErrorFoundBadEmailInFile';

Reply:
@eldy with file i just worked with, several emails were wrong, with hidden wrong caracter, then return of langs->trans() was empty, i don't know why.
The problem is that in your case (with else condition) last full error will be overwrited by 'ErrorFoundBadEmailInFile' which gives no real information about wich one is wrong.